### PR TITLE
refactor: new theme.config format

### PR DESCRIPTION
**Pull request by benlife5**
- allows the top level keys of theme.config to be any string -> allows custom organization in the TM UI
- reshapes the theme.config when dumping into tailwind.config, organized by the plugin value 
- no longer allows unlimited depth style objects, each section of the theme.config can only have direct child styles or one-deep StyleGroups if that group shares the same plugin
- classes created follow the format `[tailwindUtility]-[sectionName]-[styleName]-[subStyleName?]`
- css variables created follow the format `--[pluginName]-[sectionName]-[styleName]-[subStyleName?]`

Starter repo PR that uses this: https://github.com/YextSolutions/pages-visual-editor-starter/pull/159

Screenshot 2024-10-30 at 1 41 37 PM

### CODE REVIEW

File: src/utils/themeResolver.ts [hunk 1]
```diff
@@ -42,37 +44,40 @@ export type TailwindConfig = {
...
+      if (!output[style.plugin]) {
+        // add theme extension to tailwindConfig if it does not exist
+        output[style.plugin] = {};
+      }
 
-  for (const category in input) {
-    const categoryData = input[category];
-    output[category] = extractDefaults(categoryData.styles, category);
+      if ("default" in style) {
+        // It's a Style, give assign a class name and CSS variable
```
Comment (asanehisa) [inline src/utils/themeResolver.ts:64]: wording is confusing

**Comment by benlife5** [comment]
Having all sorts of pnpm-lock issues, going to wait for https://github.com/yext/visual-editor/pull/122 to merge, then will fix conflicts

**Review by asanehisa — APPROVED** [review]
+1

**Comment by benlife5** [comment]
Replaced with #126 because these tests kept failing

**Pull request closed without merging**